pypesto/optimize/ess/sacess.py

```diff
@@ -683,20 +692,14 @@ def run(
             # perform one ESS iteration
             ess._do_iteration()
 
-            if self._tmp_result_file:
```

Comment:
Another hyper parameter like `trace_save_iter` for ESS?

Reply:
Not really. This class is not intended to be used directly, only through `SacessOptimizer`. In that case, this will always evaluate to True. It's rather for unit testing purposes where one doesn't want/need any intermediary result files.

Reply:
Or was that a request to make saving intermediate result optional in `SacessOptimizer`?

Reply:
Yes @dweindl sorry, shaky mouse. Don't see it be super urgent to add this, but agreed, add separately.